Web/Edubase.Services/Nomenclature/NomenclatureService.cs

```diff
@@ -27,12 +28,25 @@ public string GetGovernorRoleName(eLookupGovernorRole role, eTextCase textCase =
             return name;
         }
 
-        public string GetEstablishmentsPluralName(GT groupType, eTextCase textCase = eTextCase.TitleCase)
+        public static string GetEstablishmentsPluralName(GT groupType, eTextCase textCase = eTextCase.TitleCase)
         {
-            if (groupType.OneOfThese(GT.ChildrensCentresCollaboration, GT.ChildrensCentresGroup)) return "children's centres".ToTextCase(textCase);
-            else if (groupType.OneOfThese(GT.Federation, GT.Trust)) return "schools".ToTextCase(textCase);
-            else if (groupType.OneOfThese(GT.MultiacademyTrust, GT.SingleacademyTrust, GT.SchoolSponsor)) return "academies".ToTextCase(textCase);
-            else throw new NotImplementedException($"Group type '{groupType}' is not supported for this operation");
+            switch(groupType)
+            {
+                case GT.ChildrensCentresCollaboration:
+                case GT.ChildrensCentresGroup:
+                    return "children's centres".ToTextCase(textCase);
+                case GT.Federation:
+                case GT.Trust:
+                    return "schools".ToTextCase(textCase);
+                case GT.MultiacademyTrust:
+                case GT.SingleacademyTrust:
+                case GT.SchoolSponsor:
```

Comment:
It does reduce some readability, but perhaps consider making it explicit where we are deliberately intending to fall-through on the case statements?

Something like this:

```csharp
            switch(groupType)
            {
                case GT.ChildrensCentresCollaboration:
                    // fall-through
                case GT.ChildrensCentresGroup:
                    return "children's centres".ToTextCase(textCase);
                case GT.Federation:
                    // fall-through
                case GT.Trust:
                    return "schools".ToTextCase(textCase);
                case GT.MultiacademyTrust:
                    // fall-through
                case GT.SingleacademyTrust:
                    // fall-through
                case GT.SchoolSponsor:
                    return "academies".ToTextCase(textCase);
                case GT.UmbrellaTrust:
                    // fall-through
                default:
                    throw new ArgumentOutOfRangeException(nameof(groupType),
                        $"Group type '{groupType}' is not supported for this operation");
            }
```

Reply:
Interesting point.

I think this is down to how I read case statements and the model that exists in my head for them. I prefer without the comments, I would rather copy and paste the return lines, than the commnets, but also don't like duplicated code like this.

Happy to go through this

Reply:
Feel free to resolve/disregard this suggestion and leave as-is :)